pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -1215,6 +1217,81 @@ func (p *Pod) ReclaimablePods() ([]kueue.ReclaimablePod, error) {
 	return result, nil
 }
 
+func (p *Pod) CustomWorkloadConditions(wl *kueue.Workload) ([]metav1.Condition, bool) {
+	var (
+		conditions           []metav1.Condition
+		needToUpdateWorkload bool
+	)
+	if condition, updated := p.waitingForReplacementPodsCondition(wl); condition != nil {
+		conditions = append(conditions, *condition)
+		if updated {
+			needToUpdateWorkload = true
+		}
+	}
+	return conditions, needToUpdateWorkload
+}
+
+func (p *Pod) waitingForReplacementPodsCondition(wl *kueue.Workload) (*metav1.Condition, bool) {
+	replCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadWaitingForReplacementPods)
+	replCondStatus := p.hasNotRunnableNorSucceededPods()
+
+	// Nothing to change.
+	if replCond == nil && !replCondStatus {
+		return nil, false
+	}
+
+	if replCond == nil {
+		replCond = &metav1.Condition{
+			Type:   kueue.WorkloadWaitingForReplacementPods,
+			Status: metav1.ConditionFalse,
+		}
+	}
+
+	var updated bool
+
+	if replCondStatus {
+		if replCond.Status != metav1.ConditionTrue {
+			replCond.Status = metav1.ConditionTrue
+			updated = true
+		}
+	} else {
+		if replCond.Status != metav1.ConditionFalse {
+			replCond.Status = metav1.ConditionFalse
```

Comment:
Maybe we should keep a generic reason and message here, and not update it in L1264-L1274

Reply:
Fixed